src/ui/components/RecipeEnrollmentPage.tsx

```diff
@@ -1,38 +1,129 @@
-import { ChangeEvent, FC, useState, useCallback } from "react";
-import { Form, Container, Button } from "react-bootstrap";
+import { ChangeEvent, FC, useState, useCallback, useMemo } from "react";
+import { Form, Container, Button, Modal } from "react-bootstrap";
+import { NimbusExperiment } from "@mozilla/nimbus-schemas";
 
 import { useToastsContext } from "../hooks/useToasts";
 
 const RecipeEnrollmentPage: FC = () => {
   const [jsonInput, setJsonInput] = useState("");
+  const [enrollError, setEnrollError] =
+    useState<EnrollInExperimentResult["error"]>(null);
   const { addToast } = useToastsContext();
 
+  const recipe = useMemo<NimbusExperiment | null>(() => {
+    try {
+      return JSON.parse(jsonInput) as NimbusExperiment;
+    } catch {
+      return null;
+    }
+  }, [jsonInput]);
+
+  const slug = recipe?.slug || "";
```

Comment:
We're doing this all to get the slug. Instead, lets change this slightly:

```typescript
type EnrollError = EnrollInExperimentResult["error"] & { slug: string; };
const [enrollError, setEnrollError] = useState<EnrollError | null>(null);

// line 63
setEnrollError({ slug: parsedRecipe.slug ?? "", ...result.error });
```

Since we're already parsing this to pass it off to the API, we may as well grab the slug there and only if we're parsing successfully and there is an error. otherwise, just wasted work and threading data through awkwardly.